l10n_it_account_stamp/models/invoice.py

```diff
@@ -5,9 +5,20 @@
 
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
+
+    @api.depends(
+        'invoice_line_ids.price_subtotal', 'tax_line_ids.amount',
+        'tax_line_ids.amount_rounding', 'currency_id', 'company_id', 'date_invoice',
+        'type'
+    )
+    def _compute_tax_stamp(self):
+        for invoice in self:
+            if invoice.auto_compute_stamp:
+                invoice.tax_stamp = invoice.is_tax_stamp_applicable()
+
     tax_stamp = fields.Boolean(
-        "Tax Stamp", readonly=True, states={'draft': [('readonly', False)]},
-        compute="_compute_tax_stamp", store=True)
+        "Tax Stamp", readonly=True, states={'draft': [('readonly', False)]}, default=_compute_tax_stamp)
```

Comment:
Usare come default un metodo decorato con `api.depends` mi pare non abbia senso: i default vengono eseguiti sempre e comunque.

Calcolare `tax_stamp` solo alla creazione della fattura mi sembra molto limitativo.
Probabilmente in queste modifiche si sfrutta `api.depends` per i suoi effetti analoghi a `onchange`, ma è un uso improprio:
https://github.com/odoo/odoo/blob/7a2c23c738daa23ab469f6a780bb7df69a25bec0/odoo/api.py#L212

A livello strutturale non vedo ragioni per non usare `compute`.
Penso che la domanda a cui rispondere sia "perchè il bollo scompare?".
Se è colpa di `_compute_tax_stamp` che non preserva il valore esistente di `tax_stamp`, allora bisogna provare a modificare `_compute_tax_stamp` in modo che lo faccia.

Reply:
ho provato con compute per il calcolo automatico + inverse per l'immisione a mano. Nel momento in cui viene modificato il record tramite il validate o annullamento il flag viene tolto. Non riesco a capire che cosa lo tolga. Per quello avevo proposto il default.

Reply:
`def _compute_tax_stamp(self):`
    `for invoice in self:`
    `t = invoice.tax_stamp`
Cliccando Validate, perchè `tax_stamp` sul form lo vedo `True` mentre a debug è sempre `False`
Così non riesco mai a mantenere il valore salvato

Reply:
Non è necessario l'inverse per i campi stored, basta renderli scrivibili

Reply:
@eLBati da quello che capisco io dalla documentazione :) ,https://www.odoo.com/documentation/12.0/howtos/backend.html#computed-fields-and-default-values,
un campo computed non manterrà mai il valore settato a mano. 
Abbiamo lo stesso problema e dobbiamo trovare una soluzione. 
Quella che viene in mente a noi (@Byloth ) è solo  togliere il computed al  campo e aggiungere la logica nel write e nel create.
Se ti viene in mente altro, illuminaci di immenso :)

Reply:
> Non è necessario l'inverse per i campi stored, basta renderli scrivibili

@eLBati da quello che capisco io dalla documentazione :) ,https://www.odoo.com/documentation/12.0/howtos/backend.html#computed-fields-and-default-values,
un campo computed non manterrà mai il valore settato a mano.
Abbiamo lo stesso problema e dobbiamo trovare una soluzione.
Quella che viene in mente a noi (@Byloth ) è solo togliere il computed al campo e aggiungere la logica nel write e nel create.
Se ti viene in mente altro, illuminaci di immenso :)

Reply:
Qui
https://github.com/OCA/l10n-italy/blob/519bca388ee36c9c96b6803ce1533e42c19553f4/l10n_it_fiscal_document_type/models/account_invoice.py#L56-L61
c'è un esempio

Reply:
> Qui
> https://github.com/OCA/l10n-italy/blob/519bca388ee36c9c96b6803ce1533e42c19553f4/l10n_it_fiscal_document_type/models/account_invoice.py#L56-L61
> 
> 
> c'è un esempio

Anche nel caso che citi, l'intervento manuale viene annullato da una qualsiasi operazione effettuata su uno dei depends di _set_document_fiscal_type.  

L'unica differenza è che, nel caso del documento fiscale, può avere senso alla modifica di partner_id, journal_id, type, fiscal_position_id  che la scelta venga annullata, ripristinandolo al default (a seconda delle regole conosciute o mettendolo se non ce ne sono come ([TD01] fattura), ma non in tutti i casi.

Questo comportamento, infatti, non accettabile nel caso delle spunta dell'imposta di bollo. 
Ogni volta che si aggiunge una riga di dettaglio (che è un depends) la spunta viene messa a False. 
In questo caso viene persa anche al save, quindi è impossibile indicare il bollo manualmente.
Sto cercando di capirne il motivo (nel caso del documento fiscale invece mantiene la scelta), ma non riesco a venirne a capo.